packages/calcite-components/src/components/meter/meter.tsx

```diff
@@ -0,0 +1,543 @@
+    setUpLoadableComponent(this);
+    this.calculateValues();
+    afterConnectDefaultValueSet(this, this.value);
+  }
+
+  componentDidLoad(): void {
+    setComponentLoaded(this);
+    this.updateLabels();
+  }
+
+  connectedCallback(): void {
+    connectLocalized(this);
+    connectForm(this);
+    this.resizeObserver?.observe(this.el);
+  }
+
+  disconnectedCallback(): void {
+    disconnectLocalized(this);
+    disconnectForm(this);
+    this.resizeObserver?.disconnect();
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private State/Props
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteMeterElement;
+
+  defaultValue: Meter["value"];
+
+  formEl: HTMLFormElement;
+
+  labelEl: HTMLCalciteLabelElement;
+
+  private highLabelEl: HTMLDivElement;
+
+  private labelFlipMax = 0.8;
+
+  private labelFlipProximity = 0.15;
+
+  private lowLabelEl: HTMLDivElement;
+
+  private maxLabelEl: HTMLDivElement;
+
+  private maxPercent = 100;
+
+  private meterContainerEl: HTMLDivElement;
+
+  private minLabelEl: HTMLDivElement;
+
+  private minPercent = 0;
+
+  private percentFormatting: {
+    formatter: Intl.NumberFormat;
+    locale: SupportedLocale;
+  };
+
+  private resizeObserver = createObserver("resize", () => this.resizeHandler());
+
+  private valueLabelEl: HTMLDivElement;
+
+  @State() currentPercent: number;
+
+  @State() effectiveLocale: string;
+
+  @State() highActive: boolean;
+
+  @State() highPercent: number;
+
+  @State() lowActive: boolean;
+
+  @State() lowPercent: number;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private resizeHandler(): void {
+    this.updateLabels();
+  }
+
+  private updateLabels(): void {
+    if (this.valueLabelEl) {
+      this.determineValueLabelPosition();
+    }
+    if (this.rangeLabels) {
+      this.determineVisibleLabels();
+    }
+  }
+
+  private calculateValues(): void {
+    const { min, max, low, high, value } = this;
+    const lowPercent = (100 * (low - min)) / (max - min);
+    const highPercent = (100 * (high - min)) / (max - min);
+    const currentPercent = (100 * (value - min)) / (max - min);
+
+    if (!low || low < min || low > high || low > max) {
+      this.low = min;
+    }
+    if (!high || high > max || high < low || high < min) {
+      this.high = max;
+    }
+    if (!value) {
+      this.value = min;
+    }
+
+    this.lowPercent = lowPercent;
+    this.highPercent = highPercent;
+    this.currentPercent = value ? currentPercent : 0;
+    this.lowActive = !!low && low > min && (!value || low > value) && (!high || low < high);
+    this.highActive =
+      !!high && min <= high && high < max && (!value || high > value) && (!low || high > low);
+  }
+
+  /**
+   *
```

Comment:
This can probably be dropped since the function signature is pretty expressive and typed.